### PR TITLE
CXP-1480: Update the Demo app JSON schema

### PR DESCRIPTION
**Pull request by masacc**
image

image

### CODE REVIEW

File: tests/Integration/Controller/ShowProductActionTest.php [hunk 1]
```diff
@@ -187,19 +187,52 @@ public function itDisplaysAMappedProduct(): void
...
-        $eanLabel = $foundAttributes->eq(1)->filter('.attribute__label')->text();
-        $this->assertEquals('Title', $eanLabel);
+        $label1 = $foundAttributes->eq(1)->filter('.attribute__label')->text();
+        $this->assertEquals('SKU (Stock Keeping Unit)', $label1);
 
-        $eanLabel = $foundAttributes->eq(2)->filter('.attribute__label')->text();
-        $this->assertEquals('Description', $eanLabel);
+        $label2 = $foundAttributes->eq(2)->filter('.attribute__label')->text();
+        $this->assertEquals('Product name', $label2);
+
```
Comment (hogren) [inline tests/Integration/Controller/ShowProductActionTest.php:200]: IMO, we can reuse an unique variable `$label`, or even zero variable.

File: src/Query/FetchMappedProductsQuery.php [hunk 2]
```diff
@@ -26,12 +21,13 @@ public function __construct(private readonly PimCatalogApiClient $catalogApiClie
...
     {
+        /** @var array<RawMappedProduct> $rawMappedProducts */
         $rawMappedProducts = $this->catalogApiClient->getMappedProducts($catalogId);
 
         $products = [];
         foreach ($rawMappedProducts as $rawMappedProduct) {
-            $label = isset($rawMappedProduct['title']) && '' !== $rawMappedProduct['title']
-                ? $rawMappedProduct['title']
+            $label = '' !== $rawMappedProduct['name']
+                ? $rawMappedProduct['name']
```
Comment (hogren) [inline src/Query/FetchMappedProductsQuery.php:30]: Why did you remove the `isset` test ?

**Review by hogren — APPROVED** [review]

File: src/PimApi/Model/ProductValue.php [hunk 3]
```diff
@@ -6,10 +6,13 @@
 
 class ProductValue
 {
+    /**
+     * @param string|bool|int|float|array<string>|null $value
+     */
```
[the remark this answers is not in the document]
Comment (masacc) [reply src/PimApi/Model/ProductValue.php:11]: 1. because of the new `array` type that is not enough precise in pure PHP for our static analysis tools
2. as I remember, phpstan forced me this order, I will try to be consistent

File: src/Query/FetchMappedProductsQuery.php (hunk 2 above)
Comment (masacc) [reply src/Query/FetchMappedProductsQuery.php:30]: Because the new JSON schema requires the `name` attribute to be present so we can't have any products without it.

File: tests/Integration/Controller/ShowProductActionTest.php (hunk 1 above)
Comment (hogren) [reply tests/Integration/Controller/ShowProductActionTest.php:200]: I am not sure that zero variable is a good solution for readability.

But for the first proposition, IMO it's less source of error than your solution. For each test, you need to increment the "iterator" at three places.
But, it's a detail. We can debate later about this non primordial subject.

Comment (RiffVentura) [reply tests/Integration/Controller/ShowProductActionTest.php:200]: Isn't a for-loop over an array of strings is the way ?

Comment (masacc) [reply tests/Integration/Controller/ShowProductActionTest.php]: done with a for-loop
